File: bigdime-commons/src/main/java/io/bigdime/core/commons/CollectionUtil.java
package io.bigdime.core.commons;

import java.util.Collection;

public class CollectionUtil {
	private static CollectionUtil instance = new CollectionUtil();

	private CollectionUtil() {
	}

	public static CollectionUtil getInstance() {
		return instance;
	}

	/**
	 * Simple util function to return the size of the collection.
	 * 
	 * @param collection
	 *            collection object whose size to be returned. Null collection
	 *            is accepted.F
	 * @return size of the collection if the collection is not null, return -1
	 *         otherwise
	 */
	public static int getSize(final Collection<? extends Object> collection) {
		// PECS
		if (collection == null)
			return -1;
		return collection.size();
	}

	public static boolean isEmpty(final Collection<? extends Object> collection) {
		return (collection == null || collection.isEmpty());
	}

	public static boolean isNotEmpty(final Collection<? extends Object> collection) {
		return !CollectionUtil.isEmpty(collection);
	}
}

File: bigdime-commons/src/main/scala/io/bigdime/core/commons/CollectionUtil.scala
package io.bigdime.core.commons

import java.util.Collection

/**
  * Created by neejain on 2/13/17.
  */
object CollectionUtil {
  def getSize(collection: Collection[_]): Int = {
    // PECS
    if (collection == null) return -1
    collection.size
  }

  def isEmpty(collection: Collection[_]): Boolean = collection == null || collection.isEmpty

  def isNotEmpty(collection: Collection[_]): Boolean = !isEmpty(collection)
}
